src/Avalonia.Controls.TreeDataGrid/Models/TreeDataGrid/HierarchicalRows.cs

```diff
@@ -72,6 +74,33 @@ public void Expand(IndexPath index)
             }
         }
 
+        internal void ExpandRecursive(Func<TModel, bool>? filter)
+        {
+            static void Expand(IReadOnlyList<HierarchicalRow<TModel>> rows, Func<TModel, bool>? filter)
```

Comment:
I'm wondering if it would be interesting to expose this method. We probably need to expand some nodes inside the hierarchy, so probably having this public should be interesting.

Reply:
It's already [exposed in the source](https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/pull/273/files#diff-9935eb3d0346f65b8d3a7653b52583a5a81ec6fe8142bf9ac4a4903b1c86abb9R111) which is where it should be called from. I think it was probably a mistake to expose the other expand and collapse methods in `HirearchicalRow` (to limit API surface) so I didn't want to expose any more than was necessary.

Reply:
Oh I think I understand what you're saying now. Yes, if you already have a row instance then I guess it could be useful. I can expose them if it would be useful to you.

Reply:
If [this PR is accepted](https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/pull/275), we won't need it. But I still think it's a useful API for others if one needs to recursively expand a non-root node :-).